error_handler.go

```diff
@@ -28,15 +28,15 @@ func (this *PanicErrorHandler) Fatal(from string, err error) {
 
 	logger.WithError(err).WithField("errfrom", from).Error("fatal error detected, state dump coming in stdout")
 
-	state := make(map[string]interface{})
-	state["LastSuccessfulBinlogPos"] = this.Ferry.BinlogStreamer.GetLastStreamedBinlogPosition()
-	state["LastSuccessfulPrimaryKeys"] = this.Ferry.DataIterator.CurrentState.LastSuccessfulPrimaryKeys()
-	state["CompletedTables"] = this.Ferry.DataIterator.CurrentState.CompletedTables()
+	serializedState := &SerializableState{
+		GhostferryVersion:         VersionString,
+		LastKnownTableSchemaCache: this.Ferry.Tables,
+	}
+	this.Ferry.StateTracker.Serialize(serializedState)
```

Comment:
this is not very gopher, could we pass the `Tables` and `VersionString` as arguments and get back a `SerializableState` instead?